apps/web/components/eventtype/EventSetupTab.tsx

```diff
@@ -290,22 +290,19 @@ export const EventSetupTab = (
                   setDefaultDuration(
                     selectedMultipleDuration.find((opt) => opt.value === option?.value) ?? null
                   );
-                  formMethods.setValue("length", Number(option?.value));
+                  formMethods.setValue("length", option?.value);
                 }}
               />
             </div>
           </div>
         ) : (
           <TextField
             required
-            name="length"
             type="number"
             label={t("duration")}
-            addOnSuffix={<>{t("minutes")}</>}
             defaultValue={eventType.length ?? 15}
-            onChange={(e) => {
-              formMethods.setValue("length", Number(e.target.value));
-            }}
+            {...formMethods.register("length")}
```

Comment:
Use standard RHF approach. This ensures that `length` is defined without onChange as well. But it causes length to be become a string and tRPC requires it to be a number. That has been handled by introducing zodResolver and adding a string to number transform.

Recent duration-related code checks `length` and it is  undefined if onchange hasn't occurred.

I could have fixed the duration code as well, but it seems unintuitive to have `length` be undefined.